datafusion/datasource-parquet/src/mod.rs

```diff
@@ -244,7 +242,7 @@ impl ParquetExecBuilder {
             inner: DataSourceExec::new(Arc::new(base_config.clone())),
             base_config,
             predicate,
-            pruning_predicate: parquet.pruning_predicate,
+            pruning_predicate: None, // for backwards compat since `ParquetExec` is only for backwards compat anyway
```

Comment:
Open to other suggestions (i.e. removing it). I felt like this minimizes breakage for folks still using `ParquetExec`, who are likely the same folks that want to do the least amount of work possible to upgrade.

Reply:
I filed a PR to track this
- https://github.com/apache/datafusion/issues/15950